seastar/src/timer.rs

```diff
@@ -110,4 +127,127 @@ impl SteadyClockTimer {
             sct_set_callback(self.timer.pin_mut(), boxed_callback, caller, dropper);
         }
     }
+
+    /// Sets the timer expiration time.
+    ///
+    /// It is illegal to arm a timer that has already been armed (and not disarmed by
+    /// expiration or [`SteadyClockTimer::cancel`]). In the current implementation,
+    /// this will result in an assertion failure. See [`SteadyClockTimer::rearm_until`].
+    ///
+    /// # Arguments
+    /// * `until` - The time when the timer expires.
+    pub fn arm_until(&mut self, until: SteadyClockTimePoint) {
+        seastar::assert_runtime_is_running!();
+        sct_arm_until(self.timer.pin_mut(), until);
+    }
+
+    /// Sets the timer expiration time with automatic rearming.
+    ///
+    /// It is illegal to arm a timer that has already been armed (and not disarmed by
+    /// expiration or [`SteadyClockTimer::cancel`]). In the current implementation,
+    /// this will result in an assertion failure. See
+    /// [`SteadyClockTimer::rearm_until_periodic`].
+    ///
+    /// # Arguments
+    /// * `until` - The time when the timer expires.
+    /// * `period` - Utomatic rearm duration.
+    pub fn arm_until_periodic(&mut self, until: SteadyClockTimePoint, period: SteadyClockDuration) {
```

Comment:
The `_until` part in `arm_until_periodic` suggests to me that the timer will be fired periodically _until_ some point in time. How about `arm_at_periodic`? And then changing the other functions to use `at` instead of `until`?

cc: @cvybhu @Lorak-mmk what do you think?

Reply:
I changed all `until`s to `at` for now.